src/jmespath.net/Expressions/JmesPathIdentifier.cs

```diff
@@ -16,7 +19,15 @@ public JmesPathIdentifier(string name)
         protected override JmesPathArgument Transform(JToken json)
         {
             var jsonObject = json as JObject;
-            return jsonObject?[name_];
+            return jsonObject?[name_] ?? Evaluate(name_);

```

Comment:
Function `let()` uses context to specify scoped named-values. Those values are evaluated using the `JmesPathIdentifier` expression. If the identifier is not in the current object, let’s search in the scopes stack.

Reply:
I'm not sure about the right order of the resolution. 
What about overriding? 

```
search(let({c: `other`}, &c), {"c":"z"})
```

should give "z" or "other" ? (I assume it must be other if we support overriding.

Reply:
Currently, the spec in unambiguous and says we should evaluate the current context first, before reaching to the scopes.